src/core/manager-serialize.c

```diff
@@ -23,11 +23,12 @@ int manager_open_serialization(Manager *m, FILE **ret_f) {
         return open_serialization_file("systemd-state", ret_f);
 }
 
-static bool manager_timestamp_shall_serialize(ManagerTimestamp t) {
-        if (!in_initrd())
+static bool manager_timestamp_shall_serialize(ManagerObjective o, ManagerTimestamp t) {
+        if (!in_initrd() && o != MANAGER_SOFT_REBOOT)
                 return true;
 
-        /* The following timestamps only apply to the host system, hence only serialize them there */
+        /* The following timestamps only apply to the host system (or first boot in case of soft-reboot),
+         * hence only serialize them there. */
```

Comment:
hmm? this comment seems inverted? i.e. don#t match the code?

what's the goal here? that the timestamps are updated on each soft reboot cycle or that they only tell us about the first soft reboot cycle?

The comment suggests the latter, but the code suggests the former?

I think the whole code here is a bit wrong (previously already). We really should serialize these timestamps in the initrd too (for example generator timestamps do matter in the initrd too and should not be lost in case of daemon-reload).

It's not entirely clear to me how this actually should work, I have the suspicion that what your comment said was actually right. And in code that would mean to me: drop the in_initrd() conditionalization, drop any special casing of soft-reboot. And instead skip the serialization *only* on the MANAGER_SWITCH_ROOT case.

That would mean that the timestamps would reflect the timing *after* the most recent switch-root, and that's it.

Reply:
this really should be documented in the dbus interface man page thing

Reply:
Neither: the comment and the code make it so that everything is serialized if we are not in the initrd and we are not soft-rebooting (ie, reexec/reload).
Now whether the logic about the initrd is right or not, I have no idea, and have found no issue with it - so I'll leave it as an exercise for the reader to look into it and refactor it, if needed.